core/esmf-aspect-model-document-generators/src/main/java/org/eclipse/esmf/aspectmodel/generator/diagram/AspectModelDiagramGenerator.java

```diff
@@ -180,7 +185,18 @@ private void generatePng( final String dotInput, final OutputStream output ) thr
       final Graphviz graphviz = Graphviz.fromGraph( g );
       graphviz.render( guru.nidi.graphviz.engine.Format.PNG ).toOutputStream( output );
    }
-
+   private File gnererateTmpFontFile() throws IOException {
+      File tempFontFile =  new File(TMP_FONT_FILE);
```

Comment:
1. I will make the tempFontFile variable final and will use, `System.getProperty("java.io.tmpdir")` to get the tmp path so that it can work on all OS.
2. Creating a file using `Files.createTempFile( "aspect-model-diagram", null ).toFile()` will create a file with a random name. so, rather than creating the file using the temp file create function can we create a proper file but inside tmp? so, for each request, we don't need a new file and can use an existing one?

Reply:
1. I will make the tempFontFile variable final and will use, `System.getProperty("java.io.tmpdir")` to get the tmp path so that it can work on all OS.
2. Creating a file using `Files.createTempFile( "aspect-model-diagram", null ).toFile()` will create a file with a random name. so, rather than creating the file using the temp file create function can we create a proper file but inside tmp? so, for each request, we don't need a new file and can use an existing one? 
   `final File tempFontFile = new File(System.getProperty("java.io.tmpdir")+"/aspect-model-diagram/Font-RobotoCondensed-Regular.tmp" );`

Reply:
Since the content of this file will always be the same (it's always the font) this sounds reasonable. But you would have to use `File.separator` or construct it using `java.nio.file.Path` to prevent using the platform-dependent `/`.